### PR TITLE
blog: Building a Retrieval-Augmented Generation Chatbot with SvelteKit and Xata Vector Search

### PR DESCRIPTION
**Pull request by rishi-raj-jain**
## Summary

Guest tutorial blog from a contractor we've been working with recently.

Original Notion: https://www.notion.so/reeshee/Building-a-Retrieval-Augmented-Generation-Chatbot-with-SvelteKit-and-Xata-Vector-Search-1675f97648e34d2b8b2b2dd2c5cc6eb9?pvs=4

### Blog Release Checklist

- \[ ] Clean up image sizes
- \[ ] Release on: \<mm/dd/yy>
- \[ ] Published on dev.to (connected to Xata org)
- \[ ] OG image included
- \[ ] Social media is scheduled for this post

### CODE REVIEW

**Comment by rishi-raj-jain** [comment]
@tsg

may we get a preview link for this?

File: rag-sveltekit-xata.mdx [hunk 1]
```diff
@@ -256,12 +252,12 @@ const replicate = new Replicate({
...
-2. Pass the dynamically generated context (as string) for the AI to consider before generating responses. This is done via creating a message with the role of `system` and passing our relevant context string in the `content` field.
-3. Pass the whole conversation between user and chatbot as `messages`, which contains each message with it’s relevant `content` and `role` field.
+1.  Pass instructions so that the code blocks in the markdown-like responses generated by AI are prepended with \`\`\`\`\`. This'll help us highlight them in our frontend setup via a syntax highlighter. This is done via creating a message with the role of `assistant`and passing our relevant instructions in the`content` field.
+2.  Pass the dynamically generated context (as string) for the AI to consider before generating responses. This is done via creating a message with the role of `system` and passing our relevant context string in the `content` field.
+3.  Pass the whole conversation between user and chatbot as `messages`, which contains each message with it’s relevant `content` and `role` field.
 
-```typescript
-import { experimental_buildLlama2Prompt } from 'ai/prompts'
+````typescript
```
Comment (rishi-raj-jain) [inline rag-sveltekit-xata.mdx:259]: the code blocks do have three `, right?

File: rag-sveltekit-xata.mdx [hunk 2]
```diff
@@ -299,22 +295,22 @@ Using Vercel's AI SDK we’re able to save tremendous time in setting up a react
...
-  const { input, handleSubmit, messages } = useChat()
+  const { input, handleSubmit, messages } = useChat();
 </script>
 
 <div>
   <div>
-    <form on:submit={handleSubmit}>
-      <Input
+    <form on:submit="{handleSubmit}">
+      <input
```
Comment (rishi-raj-jain) [inline rag-sveltekit-xata.mdx:309]: Here, `Input` is the component import hence shall be the Capital Case I think?

File: rag-sveltekit-xata.mdx [hunk 3]
```diff
@@ -349,22 +345,22 @@ Using `svelte-highlight` package, we’ll specifically highlight the code blocks
...
+  import typescript from 'svelte-highlight/languages/typescript';
 
-	export let lang: string
-  export let text: any
+  export let lang: string;
+  export let text: any;
 </script>
 
-<Highlight language={typescript} code={text} />
+<Highlight language="{typescript}" code="{text}" />
```
Comment (rishi-raj-jain) [inline rag-sveltekit-xata.mdx:355]: Are we sure that reactive variables wrapped in quotes work in Svelte?

**Comment by alexfrancoeur** [comment]
@rishi-raj-jain I'm going to close this PR and use this one instead: https://github.com/xataio/mdx-blog/pull/203

I can't tell you how much I appreciate you creating a PR directly! It looks like we need to update some things on our side to generate previews from forks. We'll follow up in a future PR, but for the sake of time, let's use the PR I created for further edits.

**Comment by alexfrancoeur** [comment]
Preview: https://xata-blog-pr-203.vercel.app/blog/rag-vector-sveltekit-xata

**Comment by rishi-raj-jain** [comment]
> I can't tell you how much I appreciate you creating a PR directly!

I figured that it's work for the Xata team to download Notion, rename assets and stuff. I thought I'll take that one step ahead and get it done, faster.

**Comment by rishi-raj-jain** [comment]
I am glad it did create the positive emotion (as I expected)!

**Pull request closed without merging**